pkg/venafi/firefly/firefly.go

```diff
@@ -90,20 +92,33 @@ func (c *Connector) GenerateRequest(_ *endpoint.ZoneConfiguration, req *certific
 }
 
 func (c *Connector) request(method string, resource urlResource, data interface{}) (statusCode int, statusText string, body []byte, err error) {
-	url := c.baseURL + string(resource)
+	resourceUrl := c.baseURL + string(resource)
 	var payload io.Reader
 	var b []byte
+	var values url.Values
+
+	contentType := "application/json"
+
 	if method == "POST" || method == "PUT" {
-		b, _ = json.Marshal(data)
-		payload = bytes.NewReader(b)
+		//determining if the data is type of url.Values
```

Comment:
Do this in a different function to keep code easy to read

Reply:
I prefer to keep in this way given here are gotten the payload and the values which only has meaning when the request data is of type url.Values which is expected when the request is for device flow grant given the request is needed to do using form params and these values are stored, if something is wrong, to be used in the build of the error.